### PR TITLE
Fixes #7309 - Each time we retrieve a null *PlatformOptions from AvaloniaLocator, return a default instance

### LINKED ISSUE #7309: Inconsistencies in how null vs default PlatformOptions are handled
**Describe the bug**
Explicitly specifying `.With(new X11PlatformOptions())` behaves differently than not specifying platform options at all.

**To Reproduce**

I discovered this while testing OpenGL rendering in WSL:
* In an application that uses the OpenGL demo control grabbed from ControlCatalog:
* If I have `.With(new X11PlatformOptions())` in my `Program.BuildAvaloniaApp()` method, nothing is rendered
* If I remove the `.With(new X11PlatformOptions())` line, then the teapot renders successfully...

**Expected behavior**
Explicitly passing a 'default' `X11PlatformOptions` or not passing any should result in the same behavior. In the case I describe here, the teapot should never render.

Btw, the correct way to have rendering work in my WSL setup should be to:
* Pass a non-default `X11PlatformOptions` with an empty list in `GlxRendererBlacklist`.

**Desktop:**

- OS: WSL2 (that is a Debian 10, but inside Windows 10 and with an **llvmpipe** Mesa driver)
- Version: 0.10.11

**Additional context**

What I describe above is how I discovered the bug. But the bottom line is that I think, for consistency purpose, not passing a `***PlatformOptions` should behave the same as passing a `new ***PlatformOptions()`.

I've tracked down the cause of the issue to [there](https://github.com/AvaloniaUI/Avalonia/blob/ae711466390d785c036061ce33f0b5211d0ad32c/src/Avalonia.X11/Glx/GlxDisplay.cs?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L98):

```csharp
var blacklist = AvaloniaLocator.Current.GetService<X11PlatformOptions>()?.GlxRendererBlacklist;
if (blacklist != null)
   ...
```

With this code, if we explicitely passed a `new X11PlatformOptions()`, then `llvmpipe` will be blacklisted and hence no rendering in WSL, but in the case we didn't pass anything, there is no blacklist and rendering occurs...

In (most) other places, the pattern used to retrieve platform options is like this:

```csharp
AvaloniaLocator.Current.GetService<X11PlatformOptions>() ?? new X11PlatformOptions()
```

I think this behavior should be generalized. I will soon propose a PR that does just that (for all platform options).

### PR DESCRIPTION
**Pull request by odalet**
## What does the pull request do?

All places where this kind of construct is encountered:

```csharp
var opts  = AvaloniaLocator.Current.GetService<***PlatformOptions>();
if (opts?.Something) ...
```

now look like:

```csharp
var opts  = AvaloniaLocator.Current.GetService<***PlatformOptions>() ?? new ***PlatformOptions();
if (opts.Something) ...
```

## What is the current behavior?

In some places, when we retrieve `null` platform options from Avalonia service locator, the logic further ahead:
* either makes decisions inconsistent with the behavior we'd have with a default platform options instance (as was the case in the original issue with GLX black list)
* or reimplements default behavior (thus duplicating logic). For example, in the Linux Framebuffer case, we had:

```csharp
var opts = AvaloniaLocator.Current.GetService<LinuxFramebufferPlatformOptions>();
...
.Bind<IRenderTimer>().ToConstant(new DefaultRenderTimer(opts?.Fps ?? 60))
```

This was replaced by:

```csharp
var opts = AvaloniaLocator.Current.GetService<LinuxFramebufferPlatformOptions>() ?? new LinuxFramebufferPlatformOptions();
...
.Bind<IRenderTimer>().ToConstant(new DefaultRenderTimer(opts.Fps))
```

There's no need any more to duplicate the **60 fps** default value as it is the default in a new instance of `LinuxFramebufferPlatformOptions`. In this case, the logic does not change, but it is also less subject to a later bug if the platform options defaults were to change in the future.

## What is the updated/expected behavior with this PR?

There is only one expected behavioral change:
* OpenGL rendering that was (unexpectedly) working on Linux machines with no GPU should now stop working...
* ...because the correct way to have it working is to explicitly remove **llvmpipe** from the `X11PlatformOptions.GlxRendererBlacklist`.

## How was the solution implemented (if it's not obvious)?

To sum it up, I made sure instances of WhateverPlatformOptions were never `null` and simplified the code replacing things like `opts?.Property == true` into `opts.Property`.

## Checklist

- [ ] Added unit tests (if possible)? I couldn't find any place to add tests to, nor a way to easily test this
- [ ] Added XML documentation to any related classes? Not needed
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Documentation with user documentation? Not needed

## Breaking changes

Not sure if this is considered a breaking change as obviously using OpenGL on a Linux box with only the llvmpipe Mesa driver was not supported...

## Fixed issues

Fixes #7309

### CODE REVIEW

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0017729-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Review by kekekeks — APPROVED** [review]

**Review by danwalmsley — APPROVED** [review]
